operator/v1/types_klusterlet.go

```diff
@@ -86,6 +86,22 @@ type KlusterletSpec struct {
 	// registration-agent and work-agent will use it to communicate with hub api server.
 	// +optional
 	HubApiServerHostAlias *HubApiServerHostAlias `json:"hubApiServerHostAlias,omitempty"`
+
+	// ResourceType specify QoS classes of klusterlet deployment
+	// +optional
+	ResourceType ResourceType `json:"resourceType,omitempty"`
+}
+
+type ResourceQosClass string
+
+const (
+	ResourceQosClassDefault    ResourceQosClass = "Default"
+	ResourceQosClassBestEffort ResourceQosClass = "BestEffort"
+)
+
+// ResourceType allow user override the default pod QoS classes
+type ResourceType struct {
+	Type ResourceQosClass `json:"type"`
```

Comment:
// +kubebuilder:validation:Enum=Default;BestEffort
  // +kubebuilder:default:=Default

Should add some valildation